lib/semantics/resolve-names.cc

```diff
@@ -1757,7 +1760,10 @@ static bool NeedsType(const Symbol &symbol) {
 }
 void ScopeHandler::ApplyImplicitRules(Symbol &symbol) {
   if (NeedsType(symbol)) {
-    if (isImplicitNoneType()) {
+    if (const auto *type{GetImplicitType(symbol)}) {
```

Comment:
`DeclTypeSpec` might be more clear than `auto` here.

Reply:
Done.